src/gpu/matrix/KmMatrix/KmMatrix.cpp

```diff
@@ -0,0 +1,358 @@
+/*!
```

Comment:
Again I don't think we need to have hpp/cuh and cpp/cu files. We just need the cuda files.

Reply:
This cpp is for KmMatrix, not any specific KmMatrixImpl.  We bridge KmMatrix to KmMatrixImpl (CudaKmMatrixImpl specifically) in this file. It's a generic frontend.